aws-synthetics-group/src/main/java/com/amazon/synthetics/group/TagHelper.java

```diff
@@ -0,0 +1,304 @@
+        return Sets.difference(new HashSet<>(previousTags), new HashSet<>(desiredTags));
+    }
+
+
+    /**
+     * tagResource during update
+     *
+     * Calls the service:TagResource API.
+     */
+    private ProgressEvent<ResourceModel, CallbackContext>
+    tagResource(final AmazonWebServicesClientProxy proxy, final ProxyClient<SdkClient> serviceClient, final ResourceModel resourceModel,
+                final ResourceHandlerRequest<ResourceModel> handlerRequest, final CallbackContext callbackContext, final Map<String, String> addedTags, final Logger logger) {
+        // TODO: add log for adding tags to resources during update
+        // e.g. logger.log(String.format("[UPDATE][IN PROGRESS] Going to add tags for ... resource: %s with AccountId: %s",
+        // resourceModel.getResourceName(), handlerRequest.getAwsAccountId()));
+
+        // TODO: change untagResource in the method to your service API according to your SDK
+        return proxy.initiate("AWS-Synthetics-Group::TagOps", serviceClient, resourceModel, callbackContext)
+            .translateToServiceRequest(model ->
+                Translator.tagResourceRequest(model, addedTags))
+            .makeServiceCall((request, client) -> {
+                return (AwsResponse) null;
+                // TODO: replace the return null with your invoke log to call tagResource API to add tags
+                // e.g. proxy.injectCredentialsAndInvokeV2(request, client.client()::tagResource))
+            })
+            .progress();
+    }
+
+    /**
+     * untagResource during update
+     *
+     * Calls the service:UntagResource API.
+     */
+    private ProgressEvent<ResourceModel, CallbackContext>
+    untagResource(final AmazonWebServicesClientProxy proxy, final ProxyClient<SdkClient> serviceClient, final ResourceModel resourceModel,
+                  final ResourceHandlerRequest<ResourceModel> handlerRequest, final CallbackContext callbackContext, final Set<String> removedTags, final Logger logger) {
+        // TODO: add log for removing tags from resources during update
+        // e.g. logger.log(String.format("[UPDATE][IN PROGRESS] Going to remove tags for ... resource: %s with AccountId: %s",
+        // resourceModel.getResourceName(), handlerRequest.getAwsAccountId()));
+
+        // TODO: change untagResource in the method to your service API according to your SDK
+        return proxy.initiate("AWS-Synthetics-Group::TagOps", serviceClient, resourceModel, callbackContext)
+            .translateToServiceRequest(model ->
+                Translator.untagResourceRequest(model, removedTags))
+            .makeServiceCall((request, client) -> {
+                return (AwsResponse) null;
+                // TODO: replace the return null with your invoke log to call untag API to remove tags
+                // e.g. proxy.injectCredentialsAndInvokeV2(request, client.client()::untagResource)
+            })
+            .progress();
+    }
+
+}
```

Comment:
There are a bunch of TODOs. Flagging in case if it was missed. Alright if you are going to come back to it later.

Reply:
Keeping it for another iteration on the tags part